src/infrastructure/diConfig.ts

```diff
@@ -139,9 +211,37 @@ export function registerDependencies(
 
     userRepository: asClass(UserRepository, SINGLETON_CONFIG),
     userService: asClass(UserService, SINGLETON_CONFIG),
-    userCache: asClass(UserCache, SINGLETON_CONFIG),
-    urlCache: asClass(UrlCache, SINGLETON_CONFIG),
-    configStore: asClass(ConfigStore, SINGLETON_CONFIG),
+
+    userLoader: asFunction(
+      (deps: Dependencies) => {
+        const { publisher: notificationPublisher, consumer: notificationConsumer } =
+          createNotificationPair<User>({
```

Comment:
this fancy thing uses Redis Pub/Sub in order to notify other nodes about data invalidation, in order to prevent stale data in in-memory cache.

Reply:
Fancy!